OpenSim/Common/Storage.cpp

```diff
@@ -432,43 +422,52 @@ getStateIndex(const std::string &aColumnName, int startIndex) const
     if (thisColumnIndex >= 0)
         // subtract 1 because time is included in the labels but not 
         // in the "state vector"
-        return thisColumnIndex-1;
+        return thisColumnIndex - 1;
+
+    // Assume column labels follow pre-v4.0 state variable labeling
+    // ------------------------------------------------------------
+    // Redo search with what the pre-v4.0 label might have been.
 
-    // Assume a mismatch between earlier and the latest component state variable 
-    // labeling mechanism and redo find with the just the ending substring instead
-    // of its full path name
+    // First, try just the last element of the path.
     std::string::size_type back = aColumnName.rfind("/");
     std::string prefix = aColumnName.substr(0, back);
-    std::string shortName = aColumnName.substr(back + 1, aColumnName.length() - back);
-    thisColumnIndex = _columnLabels.findIndex(shortName);
+    std::string shortName = aColumnName.substr(back + 1,
+                                               aColumnName.length() - back);
 
-    // additional checking for old coordinate state names that have been renamed
-    // <coord_name>/value and <coord_name>/speed
-    if (thisColumnIndex < 0){
-        if (shortName == "value"){
-            // old formats did not have "/value" so remove it if here
-            back = prefix.rfind("/");
-            shortName = prefix.substr(back + 1, prefix.length());
-            thisColumnIndex = _columnLabels.findIndex(shortName);
-        }
-        else if (shortName == "speed"){
-            // replace "/speed" (the latest labeling for speeds) with "_u"
-            back = prefix.rfind("/");
-            shortName = prefix.substr(back + 1, prefix.length() - back) + "_u";
-            thisColumnIndex = _columnLabels.findIndex(shortName);
-        }
-        else if (back < aColumnName.length()) {
-            // try replacing the '/' with '.' in the last connection
-            shortName = aColumnName;
-            shortName.replace(back, 1, ".");
-            back = shortName.rfind("/");
-            shortName = shortName.substr(back + 1, shortName.length() - back);
-            thisColumnIndex = _columnLabels.findIndex(shortName);
-        }
+    thisColumnIndex = _columnLabels.findIndex(shortName);
+    if (thisColumnIndex >= 0)
+        return thisColumnIndex - 1;
```

Comment:
These changes were mainly to fix a bug where the function returned `-2` instead of `-1` when the desired column was not found. I agree that the repeated code is undesirable; I'll play around with this a bit more.